runtime-library/README.md

```diff
@@ -0,0 +1,47 @@
+# IREE runtime library builder
+
+Integrators often like to link against a single runtime library, and the
+method of doing so is naturally owned by the integrator -- not necessarily
+by IREE itself (i.e. IREE will never release a full libireert.a). This
+separation of concerns is of practical importance because IREE's low level
+runtime API is fine-grained and geared towards usage via LTO style
+optimizations. This makes it inherently non-distributable, since every
+toolchain defines such features and interop differently.
+
+Also, it is often convenient during early development (of language bindings,
+etc) to simply dynamically link to something that works, even if not optimal.
+Because it cannot produce the best integration, IREE itself does not export
+a shared runtime library. However, to aid development, it can be useful
+for users to produce one.
+
+This sample illustrates both modes of use. It may eventually be included in
+the IREE repository proper as a standalone project.
+
+## Options:
+
+* `-DBUILD_SHARED_LIBS=ON` : Builds a libireert.so (or corresponding DLL/dylib)
+  for development use. Note that the low-level IREE runtime API is fine grained
+  and usage in a shared library will pessimize optimizations. Therefore, this
+  is only recommended for development or getting started.
+* `-DIREERT_ENABLE_LTO=ON` : Enables LTO if the toolchain supports it. This is
+  supported for both shared and static library builds, but for shared libraries,
+  the optimizations stop at the exported symbol boundary. As of early 2023,
+  this has the side effect of reducing the binary size by ~16%.
+
+## Using in external builds.
+
+When built, `lib/` and `include/` directories will be populated. It should
+be possible to use these with no further manipulation. While using the main
+`iree` project as a sub-project is recommended for those who can, using
+standalone headers and libraries is expected to benefit language bindings and
+other indirect uses of the API, since such tooling often interops at the
+C/library level, not the build system level.
+
+## Typical use
+
+```
+cmake -GNinja -Bbuild .
+cd build
+ninja
+./bin/ireert_test
+```
```

Comment:
It looks like this also requires that the main `iree/` repo has been checked out as a sibling to `iree-samples/`

```cmake
set(IREE_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../iree" CACHE STRING "Main IREE source")
```

That could be listed as a prerequisite, possibly also with some git hash / version hints as needed for this code to be compatible with what's in the other repo.

Reply:
Yeah, this is a top-level cmake project in its own right, and it has to be, I'm mainly landing it here for comment/discussion/unblocking. It may actually make sense as a standalone_library directory in the main iree runtime directory, and then we would direct people building out of tree bindings and such to build that.

This has no particular version sensitivity -- the things it is doing haven't changed in years, so low churn. Need a full story for specialty users and would like to have it and iterate for now.